TestVersionCheck/Passes_when_versions_are_latest is flaky

It looks like this go unit test is occasionally failing in CI:

```
=== RUN   TestVersionCheck/Passes_when_versions_are_latest
panic: runtime error: index out of range [recovered]
  panic: runtime error: index out of range
goroutine 27 [running]:
testing.tRunner.func1(0xc4203641e0)
  /home/travis/.gimme/versions/go1.10.2.linux.amd64/src/testing/testing.go:742 +0x567
panic(0x1753c00, 0x2272af0)
  /home/travis/.gimme/versions/go1.10.2.linux.amd64/src/runtime/panic.go:502 +0x24a
github.com/runconduit/conduit/pkg/version_test.TestVersionCheck.func2(0xc4203641e0)
  /home/travis/gopath/src/github.com/runconduit/conduit/pkg/version/version_test.go:31 +0xca1
testing.tRunner(0xc4203641e0, 0x19378a8)
  /home/travis/.gimme/versions/go1.10.2.linux.amd64/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
  /home/travis/.gimme/versions/go1.10.2.linux.amd64/src/testing/testing.go:824 +0x565
FAIL  github.com/runconduit/conduit/pkg/version 0.050s
```

--- a/pkg/version/version_test.go
+++ b/pkg/version/version_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"net/http"
 	"testing"
+	"time"
 
 	"github.com/linkerd/linkerd2/controller/api/public"
 	healthcheckPb "github.com/linkerd/linkerd2/controller/gen/common/healthcheck"
@@ -16,6 +17,24 @@ func TestVersionCheck(t *testing.T) {
 		fmt.Fprintf(w, "{\"version\": \"v0.3.0\"}")
 	})
 	go http.ListenAndServe("localhost:8080", nil)
+
+	// wait for HTTP server to initialize
+	readyCh := make(chan struct{})
+	go func() {
+		for {
+			_, err := http.Head("http://localhost:8080/")
+			if err == nil {
+				close(readyCh)
+				break
+			}
+		}
+	}()
+
+	select {
+	case <-readyCh:
+	case <-time.After(5 * time.Second):
+		t.Fatalf("Failed to initialize HTTP server")
+	}
 
 	t.Run("Passes when versions are latest", func(t *testing.T) {
 		version.Version = "v0.3.0"

--- a/pkg/version/version_test.go
+++ b/pkg/version/version_test.go
@@ -16,13 +16,13 @@ func TestVersionCheck(t *testing.T) {
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "{\"version\": \"v0.3.0\"}")
 	})
-	go http.ListenAndServe("localhost:8080", nil)
+	go http.ListenAndServe("localhost:23456", nil)
 
 	// wait for HTTP server to initialize
 	readyCh := make(chan struct{})
 	go func() {
 		for {
-			_, err := http.Head("http://localhost:8080/")
+			_, err := http.Head("http://localhost:23456/")
 			if err == nil {
 				close(readyCh)
 				break
@@ -40,7 +40,7 @@ func TestVersionCheck(t *testing.T) {
 		version.Version = "v0.3.0"
 		mockPublicApi := createMockPublicApi("v0.3.0")
 
-		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:8080/", "", mockPublicApi)
+		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:23456/", "", mockPublicApi)
 		checks := versionStatusChecker.SelfCheck()
 
 		expectedName := version.VersionSubsystemName
@@ -73,7 +73,7 @@ func TestVersionCheck(t *testing.T) {
 		version.Version = "v0.1.1"
 		mockPublicApi := createMockPublicApi("v0.3.0")
 
-		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:8080/", "", mockPublicApi)
+		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:23456/", "", mockPublicApi)
 		checks := versionStatusChecker.SelfCheck()
 
 		expectedStatus := healthcheckPb.CheckStatus_FAIL
@@ -91,7 +91,7 @@ func TestVersionCheck(t *testing.T) {
 		version.Version = "v0.3.0"
 		mockPublicApi := createMockPublicApi("v0.1.1")
 
-		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:8080/", "", mockPublicApi)
+		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:23456/", "", mockPublicApi)
 		checks := versionStatusChecker.SelfCheck()
 
 		expectedStatus := healthcheckPb.CheckStatus_FAIL
@@ -109,7 +109,7 @@ func TestVersionCheck(t *testing.T) {
 		version.Version = "customversion"
 		mockPublicApi := createMockPublicApi("customversion")
 
-		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:8080/", "customversion", mockPublicApi)
+		versionStatusChecker := version.NewVersionStatusChecker("http://localhost:23456/", "customversion", mockPublicApi)
 		checks := versionStatusChecker.SelfCheck()
 
 		for _, check := range checks {

